f5_os_test/infrastructure.py

```diff
@@ -14,23 +14,144 @@
+    for sn in subnets:
+        if 'client-v4' in sn['name']:
+            lbconf = {'vip_subnet_id': sn['id'],
+                      'tenant_id':     sn['tenant_id'],
+                      'name':          'testlb_01'}
+    activelb =\
+        nclientmanager.create_loadbalancer({'loadbalancer': lbconf})
+    return nclientmanager, activelb
+
+
+@pytest.fixture
+def setup_with_listener(setup_with_loadbalancer):
+    nclientmanager, activelb = setup_with_loadbalancer
+    listener_config =\
+        {'listener': {'name': 'test_listener',
+                      'loadbalancer_id': activelb['loadbalancer']['id'],
+                      'protocol': 'HTTP',
+                      'protocol_port': 80}}
+    listener = nclientmanager.create_listener(listener_config)
+    return nclientmanager, listener
+
+
+@pytest.fixture
+def setup_with_pool(setup_with_listener):
+    nclientmanager, activelistener = setup_with_listener
+    pool_config = {'pool': {
+                   'name': 'test_pool_anur23rgg',
+                   'lb_algorithm': 'ROUND_ROBIN',
+                   'listener_id': activelistener['listener']['id'],
+                   'protocol': 'HTTP'}}
+    pool = nclientmanager.create_lbaas_pool(pool_config)
+    return nclientmanager, pool
+
+
+@pytest.fixture
+def setup_with_pool_member(setup_with_pool):
+    nclientmanager, activepool = setup_with_pool
+    pool_id = activepool['pool']['id']
+    for sn in nclientmanager.list_subnets()['subnets']:
+        if 'server-v4' in sn['name']:
+            address = sn['allocation_pools'][0]['start']
+            subnet_id = sn['id']
+            break
+
+    member_config = {'member': {
+                     'subnet_id': subnet_id,
+                     'address': address,
+                     'protocol_port': 80}}
+    member = nclientmanager.create_lbaas_member(pool_id, member_config)
+    return nclientmanager, activepool, member
+
+
+@pytest.fixture
+def setup_with_healthmonitor(setup_with_pool_member):
+    nclientmanager, activepool, activemember = setup_with_pool_member
+    monitor_config = {'healthmonitor': {
+                      'delay': 3,
+                      'pool_id': activepool['pool']['id'],
+                      'type': 'HTTP',
+                      'timeout': 13,
+                      'max_retries': 7}}
+    healthmonitor = nclientmanager.create_lbaas_healthmonitor(monitor_config)
+    return nclientmanager, healthmonitor, activepool, activemember
+
+
+@pytest.fixture
+def get_auth_token(keystoneclientmanager):
+    token_id = keystoneclientmanager.auth_ref['token']['id']
+    return token_id
+
+
+@pytest.fixture
+def heatclientmanager(heatclient_pollster, get_auth_token, symbols):
+    '''Heat client manager fixture.'''
+    config_dict = {
+        'endpoint': symbols.heatclient_url,
+        'token':    get_auth_token
+    }
+    return heatclient_pollster(**config_dict)
+
+
+@pytest.fixture
+def keystoneclientmanager(symbols, keystoneclient_pollster):
```

Comment:
I suggest we make "symbols" be the last positional parameter in all cases where that's feasible (doesn't conflict with some overriding policy).   Waddya' think?